app/models/audit/AuditTaskTable.scala

```diff
@@ -321,183 +339,168 @@ object AuditTaskTable {
   }
 
   /**
-   * get a new task for the user
+   * Get a new task for the user. This is called if the user is not already assigned a region.
    *
    * Reference for creating java.sql.timestamp
    * http://stackoverflow.com/questions/308683/how-can-i-get-the-current-date-and-time-in-utc-or-gmt-in-java
    * http://alvinalexander.com/java/java-timestamp-example-current-time-now
    *
-   * Subqueries in Slick
-   * http://stackoverflow.com/questions/14425844/why-does-slick-generate-a-subquery-when-take-method-is-called
-   * http://stackoverflow.com/questions/14920153/how-to-write-nested-queries-in-select-clause
-   *
    * @param user User ID.
    * @return
    */
   def selectANewTask(user: UUID): NewTask = db.withSession { implicit session =>
```

Comment:
Here is the start of the functions that have been changed substantially, and should be tested.